### DISCUSSION TITLE
will we add a hook to help creating multiple event listener?

### DISCUSSION THREAD

**Discussion by jukrb0x** [mantinedev/mantine#2964] [Feature requests]
> `import { useEventListener } from '@mantine/hooks';`

the current `useEventListener` only create one event listener and return `ref` , if I want to make multiple event listeners on an element (e.g. `mouseup` and `mousedown`), I would implement my own hook or do it with `useEffect()`, is there any plan for new hook like `use-event-listeners` ?

**Comment by rtivital** [comment]
You can use as many `useEventListener` hooks in your component as you need

**Reply by jukrb0x** [reply]
hi, thanks for your reply. after I dive into [source code](https://github.com/mantinedev/mantine/blob/master/src/mantine-hooks/src/use-event-listener/use-event-listener.ts), it looks like I can only bind a `ref` returned by `useEventListener` to the element, instead of this, I would like an approach to bind multiple event listeners to a `ref` (element), an example from [ahooks](https://ahooks.js.org/hooks/use-event-listener) providing here:
```javascript
import React, { useState, useRef } from 'react';
import { useEventListener } from 'ahooks';

export default () => {
  const [value, setValue] = useState(0);
  const ref = useRef(null);

  useEventListener(
    'click',
    () => {
      setValue(value + 1);
    },
    { target: ref },
  );

  return (
    <button ref={ref} type="button">
      You click {value} times
    </button>
  );
};
```

so I can add on more event listeners just like `'click'`, to the same `ref`:
```javascript
  useEventListener(
    'mousedown',
    () => {
      console.log("yeah!");
    },
    { target: ref },
  );
```

the current approach in mantine is returning a `ref` with corresponding event listener:
```javascript
  const ref = useRef<T>();

  useEffect(() => {
    if (ref.current) {
      ref.current.addEventListener(type, listener, options);
      return () => ref.current?.removeEventListener(type, listener, options);
    }
    return undefined;
  }, [listener, options]);

  return ref;
```

----
In fact, in my use case, I write a custom hook to satisfy the need:

```javascript
import { useEffect, useRef, useState } from 'react';

export const useResizer = () => {
    const [isResizing, setIsResizing] = useState(false);
    const ref = useRef<HTMLElement>();

    useEffect(() => {
        if (ref.current) {
            ref.current?.addEventListener('mousedown', () => {
                console.log("resizing");
                setIsResizing(true);
            });
            ref.current?.addEventListener('mouseup', () => {
                console.log("not resizing");
                setIsResizing(false);
            });
            return () => {
                ref.current?.removeEventListener('mousedown', () => {
                    console.log("resizing (cleanup)");
                    setIsResizing(true);
                });
                ref.current?.removeEventListener('mouseup', () => {
                    console.log("not resizing (cleanup)");
                    setIsResizing(false);
                });
            };
        }
    }, ['mousedown', 'mouseup']);
    return [isResizing, ref];
};

```

so I am wondering if mantine can provide such multi-event listener hooks for convenience?

**Reply by rtivital** [reply]
You can use `useMergedRef` hook to merge refs

**Reply by jukrb0x** [reply]
thanks a lot!
